### PR TITLE
Fixed bugs, simplified tags model, removed filter tag input

### PR DESCRIPTION
**Pull request by andronz**

### CODE REVIEW

File: HealthMonitoring.SelfHost/Controllers/EndpointsController.cs [hunk 1]
```diff
@@ -82,14 +82,23 @@ public IEnumerable<EndpointDetails> GetEndpoints()
...
+        public IHttpActionResult PostUpdateEndpointTags(Guid id, [FromBody]string[] tags)
         {
-            model.ValidateModel();
-            _endpointRegistry.UpdateEndpointTags(id, model.Tags);
-            return Ok(EndpointDetails.FromDomain(_endpointRegistry.GetById(id)));
+            tags.CheckForUnallowedSymbols();
+            try
+            {
+                if(_endpointRegistry.TryUpdateEndpointTags(id, tags))
+                    return Ok(EndpointDetails.FromDomain(_endpointRegistry.GetById(id)));
```
Comment (Suremaker) [inline HealthMonitoring.SelfHost/Controllers/EndpointsController.cs]: do we need to return the endpoint here?

Comment (williamlord) [inline HealthMonitoring.SelfHost/Controllers/EndpointsController.cs]: also PUT is create not POST, and then can return 204 no content

File: HealthMonitoring.SelfHost/Controllers/EndpointsController.cs [hunk 2]
```diff
@@ -82,14 +82,23 @@ public IEnumerable<EndpointDetails> GetEndpoints()
...
+                return NotFound();
+            }
+            catch (Exception e)
```
Comment (williamlord) [inline HealthMonitoring.SelfHost/Controllers/EndpointsController.cs]: what exceptions are we expecting here?
if it's expected it should be a particular type eg. mycustomexception, or if its unexpected should be 5xx not 4xx?

File: HealthMonitoring.AcceptanceTests/Scenarios/Endpoint_registration.cs [hunk 3]
```diff
@@ -76,13 +77,13 @@ public void Deleting_inexistent_endpoint()
...
         [Scenario]
-        public void UpdatingEndpointTags()
+        public void Updating_endpoint_tags()
         {
             Runner.RunScenario(
                 _ => Given_a_monitor_api_client(),
                 _ => Given_endpoint_with_name_address_group_and_monitor_is_registered("my_name", "http://localhost:3333/status", "my_group", MonitorTypes.HttpJson),
                 _ => When_client_requests_tags_updating_via_url($"api/endpoints/{_identifier}/tags", new [] {"tag1", "tag2"}),
-                _ => Then_endpoint_tags_should_be_updated(new[] { "tag1", "tag2" })
+                _ => Then_status_should_be_returned(HttpStatusCode.OK)
```
Comment (Suremaker) [inline HealthMonitoring.AcceptanceTests/Scenarios/Endpoint_registration.cs]: I am going to merge the change, but there should be more steps like:
_ => When_client_requests_endpoint_details(),
_ => Then_the_endpoint_tags_should_be_TAGS("tag1","tag2")

void Then_the_endpoint_tags_should_be_TAGS(params string[]tags)

**Pull request merged**